apps/analytics/views.py

```diff
@@ -88,3 +90,27 @@ def get_submission_count(request, challenge_pk, duration):
     submission_count = SubmissionCount(submission_count)
     serializer = SubmissionCountSerializer(submission_count)
     return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET', ])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail, IsChallengeCreator))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_last_submission_time(request, challenge_pk, challenge_phase_pk, submission_by):
+    challenge = get_challenge_model(challenge_pk)
+
+    challenge_phase = get_challenge_phase_model(challenge_phase_pk)
+
+    # To get the last submission time by a user in a challenge phase.
+    if submission_by == 'user':
```

Comment:
Why do we need `submitted_by` in the URL? Is there a specific use for this that I am missing here?